packages/react-native/ReactAndroid/src/main/java/com/facebook/react/bridge/UiThreadUtil.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.facebook.react.bridge
+
+import android.os.Handler
+import android.os.Looper
+import com.facebook.react.common.build.ReactBuildConfig
+ 
+/** Utility for interacting with the UI thread. */
+public object UiThreadUtil {
+ 
+    @Volatile private var sMainHandler: Handler? = null
+ 
+    private val mainHandler: Handler
+        get() {
+             if (sMainHandler == null) {
+                 synchronized(this) {
+                     if (sMainHandler == null) {
+                         sMainHandler = Handler(Looper.getMainLooper())
+                     }
+                 }
+             }
+             return sMainHandler!!
+        }
+ 
+    /** Exposed for Java interop (e.g. Java calls to `UiThreadUtil.getUiThreadHandler()`) */
+    @JvmStatic
+    public fun getUiThreadHandler(): Handler {
+         return mainHandler
+    }
+ 
+    /** @return `true` if current thread is the UI thread. */
+    @JvmStatic
+    public fun isOnUiThread(): Boolean {
+         return Looper.getMainLooper().thread == Thread.currentThread()
+    }
```

Comment:
Could you please use expression body syntax? 

Like this:

```suggestion
    @JvmStatic
    public fun isOnUiThread(): Boolean = Looper.getMainLooper().thread == Thread.currentThread()
```

Same for the others that just return.

Reply:
Done